### PR TITLE
Electron native modals plugin

### PR DESCRIPTION
**Pull request by techiediaries**
Add the modals plugin for Electron to support functionalities not available and can't be implemented in the web plugin:

For alert() and confirm(): It's not possible to set title and buttons title in the web plugin 
 
showActions() is the same

prompt() is not supported in Electron so I added an implementation using Ionic Modal controller. The modal component doesn't work when using Ionic components for interface? but works with standard html elements

### CODE REVIEW

**Comment by mlynch** [comment]
Awesome! We will review this week

**Comment by techiediaries** [comment]
@mlynch Any feedback about this?

File: electron/src/plugins.ts [hunk 1]
```diff
@@ -7,13 +7,14 @@ export * from '@capacitor/core/dist/esm/web/camera';
 export * from '@capacitor/core/dist/esm/web/geolocation';
 export * from '@capacitor/core/dist/esm/web/device';
 export * from '@capacitor/core/dist/esm/web/share';
-export * from '@capacitor/core/dist/esm/web/modals';
+//export * from '@capacitor/core/dist/esm/web/modals';
```
Comment (jcesarmobile) [inline electron/src/plugins.ts]: Don't leave commented code behind, you can delete it for good.

File: electron/src/electron/modals.ts [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+export class ModalsPluginElectron extends WebPlugin implements ModalsPlugin {
+  resolve: any;
+  constructor() {
+    super({
+      name: 'Modals',
+      platforms: ['electron']
+    });
+  }
+  async alert(options: AlertOptions): Promise<void> {
+    //window.alert(options.message); we can't set title and button title
```
Comment (jcesarmobile) [inline electron/src/electron/modals.ts]: When I read this, it doesn't mean anything to me, so better remove comments like this

File: electron/src/electron/modals.ts [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+    const alert = (message: string, title: string = '') =>
+    {    
+        
+        let buttons = [options.buttonTitle || 'OK']
+        dialog.showMessageBox(getCurrentWindow(), {message, title, buttons});       
+    }
+        alert(options.message, options.title);
+        return Promise.resolve();
+  }
+  //window.prompt is and will not be supported in Electron. Not working with Ionic components?
```
Comment (jcesarmobile) [inline electron/src/electron/modals.ts]: First part of the comment might be good to know, but the second part, not sure what do you want to say with Not working with Ionic components?. Better not leave behind this kind of comments.

File: electron/src/electron/modals.ts [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+  async prompt(options: PromptOptions): Promise<PromptResult> {
+    //const val = window.prompt(options.message, options.inputPlaceholder || '');
```
Comment (jcesarmobile) [inline electron/src/electron/modals.ts]: Same as before, not leave commented code behind.

File: electron/src/electron/modals.ts [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+    });
+  } 
+  async confirm(options: ConfirmOptions): Promise<ConfirmResult> {
+
+    const confirm = (message: string, title: string='') =>
+    {
+      let buttons = [options.okButtonTitle || 'OK' , options.cancelButtonTitle || 'Cancel']
+      return !dialog.showMessageBox(getCurrentWindow(), {message, title, buttons});
+    }
+    //const val = window.confirm(options.message);
```
Comment (jcesarmobile) [inline electron/src/electron/modals.ts]: Again, don't leave commented code behind

**Comment by techiediaries** [comment]
I've added the requested changes. Sorry about the comments. It's just a habit and also looking for any suggestions.

**Comment by jcesarmobile** [comment]
Don't worry about it.
Thanks for the PR, merged!

**Comment by techiediaries** [comment]
Thanks!

**Pull request merged**